### ISSUE TITLE
Is LLVMDependenceGraph not public on purpose?

### ISSUE THREAD

**Issue by gussmith23** [mchalupa/dg#187]
As you might have figured out, I'm totally new to using dg. I'm also new to the concepts of dependency graphs, slicing, and points-to analysis, so apologies if I'm misusing terms. 

I'm interested in using dg for building dependency graphs of my programs. I don't need any of the other capabilities as far as I know -- I simply want to build a dependency graph and walk through it looking for patterns (e.g. load-load-operation-store). My assumption was that LLVMDependenceGraph would be the class to use to do this -- however, I just realized that its header file is not actually installed on `make install`. Is this intentional? Is LLVMDependenceGraph simply an internal piece of implementation that we are not meant to use?

If I'm misjudging what dg is used for, please tell me!

**Comment by mchalupa** [comment]
It is really the right class. The problem with dg is that we haven't used it this way much. We use it as a submodule where we have access to all files, so we never run into these problems. As a result, there can be bugs like this one. I just push the fix, but feel free to create a PR with fixes in the future!

**Comment by gussmith23** [comment]
Thanks for the quick patch!

Unfortunately, I'm still having issues with some header files not being exported. Node.h isn't found; when i change src/CMakeLists.txt to export that, then another file isn't found, and so on. I could keep going, but I'm worried that soon enough I'll have included everything! So far, I've had to make the following modifications:

```
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index a77c7cf..1f70d0a 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -65,11 +65,13 @@ install(TARGETS LLVMdg LLVMpta PTA RD

 install(FILES
        ADT/Queue.h
+        ADT/DGContainer.h
        DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/llvm-dg/ADT/)
 install(FILES
        analysis/Offset.h
        analysis/SCC.h
        analysis/SubgraphNode.h
+        analysis/Analysis.h
        DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/llvm-dg/analysis/)
 install(FILES
        analysis/PointsTo/PointerAnalysis.h
@@ -94,5 +96,11 @@ install(FILES
        llvm/analysis/PointsTo/PointerSubgraphValidator.h
        llvm/analysis/PointsTo/PointsTo.h
        DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/llvm-dg/llvm/analysis/PointsTo/)
+install(FILES
+        Node.h
+        DGParameters.h
+        BBlock.h
+        DependenceGraph.h
+       DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/llvm-dg/)

 endif(LLVM_DG)
```

**Comment by gussmith23** [comment]
I'm also getting some mysterious errors which look like this:
```
In file included from /home/mdl/hfs5022/2018-spring-llvm-passes/dg-test/dg-test.cc:6:0:
/home/mdl/hfs5022/usr/local/include/llvm-dg/llvm/LLVMDependenceGraph.h:8:2: error: #error "Need LLVM"
 #error "Need LLVM"
  ^
/home/mdl/hfs5022/usr/local/include/llvm-dg/llvm/LLVMDependenceGraph.h:12:2: error: #error "Need CFG enabled"
 #error "Need CFG enabled"
  ^
```
I can only assume these are from some missing `#define`s.

**Comment by mchalupa** [comment]
Yes, those are just missing #define's that were meant to be defined by cmake once we have cmake package for dg.

**Comment by gussmith23** [comment]
So should I just continue using cmake to install the needed header files as I mentioned in [this previous comment](https://github.com/mchalupa/dg/issues/187#issuecomment-372147757)? And then I can hopefully just define those missing defines somewhere in the cmake to clear up those errors?

**Comment by mchalupa** [comment]
Yes.

**Comment by gussmith23** [comment]
I fixed it with the above linked commit (1628a33) in my fork. Thanks!
